File: leetcode/383.ransom-note/ts/canconstruct.test.ts
import { test, expect } from "vitest"
import canConstruct from "./canconstruct.js"

interface testcase {
    ransomNote: string,
    magazine: string,
    want: boolean
}

test("canconstruct", () => {
    const tcs: testcase[] = [
        { ransomNote: "a", magazine: "b", want: false },
        { ransomNote: "aa", magazine: "ab", want: false },
        { ransomNote: "aa", magazine: "aab", want: false },
    ]

    for (const tc of tcs) {
        expect(canConstruct(tc.ransomNote, tc.magazine) === tc.want)
    }
})

File: leetcode/383.ransom-note/go/canconstruct_test.go
package canconstruct

import "testing"

type testcase struct {
	ransomNote string
	magazine   string
	want       bool
}

func TestCanConstruct(t *testing.T) {
	tcs := []testcase{
		{ransomNote: "a", magazine: "b", want: false},
		{ransomNote: "aa", magazine: "ab", want: false},
		{ransomNote: "aa", magazine: "aab", want: true},
	}

	for _, tc := range tcs {
		t.Run("canconstruct", func(t *testing.T) {
			got := canConstruct(tc.ransomNote, tc.magazine)
			if got != tc.want {
				t.Fatalf("canconstruct, want: %v, got: %v", tc.want, got)
			}
		})
	}
}
